Documentation/developers/codegen.rst

```diff
@@ -0,0 +1,62 @@
+Polycube Automatic Code Generation Tools
+========================================
+
+Generating a service stub
+-------------------------
+
+``pyang`` and ``swagger-codegen`` are used to generate a service stub using the YANG datamodel.
+The former parses the YANG file and creates an intermediate JSON that is compliant with the `OpenAPI specifications <https://swagger.io/specification/>`_; the second starts from the latter file and creates a C++ code skeleton that actually implements the service. Among the services that are automatically provided, the C++ skeleton handles the case in which a complex object is requested, such as the entire configuration of the service, which is returned by disaggregating the big request into a set of smaller requests for each *leaf* object.
+Hence, this leaves to the programmer only the responsibility to implement the interaction with *leaf* objects.
+
+The easiest way to generate a stub is to use the ``polycubenetwork/polycube-codegen`` docker image.
+This contains ``pyang``, ``swagger-codegen`` and a script to invoke them.
+
+In order to run the image, it is needed to mount a volume into ``/polycube-base-datamodels`` where the base datamodels are located on the host (by default ``/usr/local/include/polycube/datamodel-common/``), it is also needed to mount a volume to share the input yang model and the output folder.
+
+::
+
+  export POLYCUBE_BASEMODELS=<path to base models usually /usr/local/include/polycube/datamodel-common/>
+  docker pull polycubenetwork/polycube-codegen
+   docker run -it --user `id -u` \
+    -v $POLYCUBE_BASEMODELS:/polycube-base-datamodels \
+    -v <input yang>:/input \
+    -v <output folder in host>:/output \
+    polycubenetwork/polycube-codegen \
+    -i /input/<yang datamodel> \
+    -o /output/<output folder>
+
```

Comment:
Since it's possible our users are not so familiar with docker syntax, I would propose to add also an explicit example like this. What do you think @mauriciovasquezbernal ?

```
export POLYCUBE_BASEMODELS="/usr/local/include/polycube/datamodel-common/"
docker pull polycubenetwork/polycube-codegen
docker run -it --user `id -u` \
  -v $POLYCUBE_BASEMODELS:/polycube-base-datamodels \
  -v $HOME/datamodels/input:/input \
  -v $HOME/datamodels/output:/output \
  polycubenetwork/polycube-codegen \
  -i /input/iptables.yang \
  -o /output/iptables_generated_code
```